uaclient/serviceclient.py

```diff
@@ -68,11 +68,15 @@ def request_url(
                 timeout=self.url_timeout,
             )
         except error.URLError as e:
-            if hasattr(e, "read"):
+            body = None
+            if hasattr(e, "body"):
+                body = e.body
```

Comment:
@orndorffgrant @CalvoM  just to be sure that I understand this change, there are other exception that have their content on the body attribute right ?

Reply:
@lucasmoura 
We added to `util.readurl` to log the response of the HTTPError and `read` can only be called once on an IO stream (`seek(0)` works on normal files, not sure if it works on an http response). Without this change, this error handling code would fail because there is nothing to `read()`.
So now, when we call `read` in `util.readurl`, we attach the content to the response/error as the new `body` attribute. Then we check for that attribute first here before trying to `read`